x-pack/plugin/ent-search/src/test/java/org/elasticsearch/xpack/application/connector/syncjob/ConnectorSyncJobIndexServiceTests.java

```diff
@@ -805,18 +806,18 @@ public void testUpdateConnectorSyncJobIngestionStats() throws Exception {
         Instant requestLastSeen = request.getLastSeen();
         Map<String, Object> metadata = request.getMetadata();
 
-        Long deletedDocumentCountAfterUpdate = (Long) syncJobSourceAfterUpdate.get(
+        Long deletedDocumentCountAfterUpdate = ((Number) syncJobSourceAfterUpdate.get(
```

Comment:
I wouldn't have expected us to have any type changes as part of defining the index as a system index.

Reply:
Why was this change necessary?

Reply:
This is because of the discrepancy between the Connector protocol and the Connectors API code. It's an issue that came to light due to introducing system index mappings to unit tests. To fix this involves altering a lot of files, so I think it's better to do separately. Tracking on [this issue](https://github.com/elastic/search-team/issues/9131).